### PR TITLE
vuln(fastify): CVE-2018-3711 Fastify denial-of-service vulnerability …

### PR DESCRIPTION
**Pull request by lirantal**
Fastify denial-of-service vulnerability with large JSON payloads.

### CODE REVIEW

File: 364.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  "publish_date": "2018-01-24",
+  "cves": [
+    "CVE-2018-3711"
+  ],
+  "vulnerable_versions": "<=0.37.0",
+  "patched_versions": ">0.38.0",
+  "slug": "fastify_remote-dos",
+  "overview": "Fastify prior to 0.37.0 is vulnerable to a denial-of-service attack by sending a request with Content-Type set to application/json and a very large payload.",
+  "recommendation": "Update to version 0.38.0 or later.",
+  "references": "- MISC:https://github.com/mqttjs/MQTT.js/commit/403ba53b838f2d319a0c0505a045fe00239e9923\n- MISC:https://github.com/mqttjs/MQTT.js/releases/tag/v2.15.0\n- JVN:JVN#45494523\n- URL:https://jvn.jp/en/jp/JVN45494523/index.html",
```
Comment (vdeturckheim) [inline 364.json]: @lirantal can you add a link to the public report on H1 ?

Comment (mcollina) [reply 364.json]: this link is wrong.

File: 364.json [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  "updated_at": "2018-01-24",
+  "title": "Fastify denial-of-service vulnerability with large JSON payloads",
+  "author": "nwoltman (https://github.com/nwoltman)",
+  "module_name": "fastify",
+  "publish_date": "2018-01-24",
+  "cves": [
+    "CVE-2018-3711"
+  ],
+  "vulnerable_versions": "<=0.37.0",
+  "patched_versions": ">0.38.0",
```
Comment (vdeturckheim) [inline 364.json]: Should be `>=`

File: 364.json (hunk 1 above)
Comment (lirantal) [reply 364.json]: yep, thanks.

**Comment by lirantal** [comment]
pending merge until I get the public report from H1 and will update the links accordingly.

**Comment by lirantal** [comment]
updated the references, are they set ok?

**Comment by vdeturckheim** [comment]
@lirantal https://hackerone.com/reports/303632 is not public yet.

**Comment by lirantal** [comment]
I would like to do it so I cover all the ins and outs of the process. That is essentially just a media announcement and through H1, right?
We basically got approval to disclose it on 24th and I sent already request through H1 to publicly disclose it which is waiting's the hacker's and maintainer's approval.

**Comment by vdeturckheim** [comment]
lgtm, I usually force disclosure myself when I know all parties agree on it. Your call here :)

**Comment by lirantal** [comment]
Right, will disclose it later today probably 👍

**Pull request merged**